### PR TITLE
Fix #145, no correct Rust compilation

### PR DESCRIPTION
**Pull request by RubenChM**
Fix for [#145](https://github.com/p-j-smith/lipyphilic/issues/145). With this conda will build different packages 
for at least the python versions 3.10 and 3.11, with the correct compiled version of lipyferric.

Also with [numpy 2.0](https://numpy.org/devdocs/release/2.0.0-notes.html), there has been some breaking changes that makes the importing of lipyphilic fail like:
` 
  File "/home/rchaves/miniforge3/envs/test/lib/python3.12/site-packages/numpy/__init__.py", line 400, in __getattr__
    raise AttributeError(AttributeError: np.NaN was removed in the NumPy 2.0 release. Use np.nan instead.
`

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by RubenChM** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -7,32 +7,33 @@ package:
...
   script:
     - {{ PYTHON }} -m pip install . -vv
     - cd rust
     - cargo-bundle-licenses --format yaml --output ../THIRDPARTY.yml
+  skip: true  # [ py==39 ]
 
 requirements:
   build:
-    - python                                 # [build_platform != target_platform]
+    - python >=3.10,<3.12                    # [build_platform != target_platform]
```
Comment (p-j-smith) [inline recipe/meta.yaml]:
```suggestion
    - python >=3.10,<3.13                    # [build_platform != target_platform]
```
Is there an issue with building with 3.12?

**Review by p-j-smith — CHANGES_REQUESTED** [review]
Thanks for fixing this @RubenChM!

Just one question around the Python versions used for building. Also, would you like to add yourself to the list of maintainers for this feedstock (no worries if not)

File: recipe/meta.yaml (hunk 1 above)
Comment (RubenChM) [reply recipe/meta.yaml]: There is not! 
It is actually building for 3.12 and it even passed the [tests](https://github.com/conda-forge/lipyphilic-feedstock/pull/14/checks?check_run_id=33900768761) for every OS.
It took me some time to realize the build variants are the ones created at .ci_support with `conda smithy rerender` and this add variants from a few places:
```
Adding in variants from internal_defaults
Adding in variants from /home/rchaves/.cache/conda-smithy/conda_build_config.yaml
Adding in variants from /home/rchaves/repo/lipyphilic-feedstock/recipe/conda_build_config.yaml
Adding in variants from argument_variants
```
I am not sure why it worked with the `<3.12`  in the build requirements, but it is nice having it at least for clarity.

**Comment by RubenChM** [comment]
Yes, you can add me! I am doing some membrane analysis right now, so I may help with the library.

**Review by p-j-smith — APPROVED** [review]
> Yes, you can add me! I am doing some membrane analysis right now, so I may help with the library.

oh cool 😄 

Thanks again for the fix!

**Pull request merged**